src/server.ts

```diff
@@ -107,7 +108,12 @@ async function search(
 		return;
 	}
 
-	const criteriaStr = inspect(criteria);
+	const criteriaStr = criteria.infoHash
+		? inspect(criteria).replace(
+				/infoHash: '.*'/i,
+				`infoHash: '${sanitizeInfoHash(criteria.infoHash)}'`,
+			)
+		: inspect(criteria);
```

Comment:
why do info hashes need to be sanitized? also, can this be done by just replacing infoHash?

```js
inspect(criteria).replace(
  criteria.infoHash, 
  sanitizeInfoHash(criteria.infoHash)
)
```

Reply:
When logs are shared, full infohashes are not needed and add what is usually unnecessary length to the logging strings.

Trackers and users both have voiced a desire to not have them shared in full, as well.

The first bit of characters generally provides the distinction that would be required to find exactly which torrent, my preference was the first and last 5 with an ellipse, but @ShanaryS thought this was fine to keep as the sanitization.